isovar/locus_reads.py

```diff
@@ -29,17 +28,34 @@
 )
 from .common import list_to_string
 from .dataframe_builder import DataFrameBuilder
-
+from .value_object import ValueObject
 
 logger = logging.getLogger(__name__)
 
-class LocusRead(namedtuple("LocusRead", [
+class LocusRead(ValueObject):
+    __slots__ = [
         "name",
         "sequence",
         "reference_positions",
         "quality_scores",
         "base0_read_position_before_variant",
-        "base0_read_position_after_variant"])):
+        "base0_read_position_after_variant"
+    ]
+
+    def __init__(
```

Comment:
To reprise the previous conversation a bit: I'm generally in favor of explicit constructors and using normal Python objects, and less in favor of writing new types of Python objects that outsource functionality to slots etc. This seems like it's a solution in search of a problem and it complicates things; I'm still not sure of what the clear win is with the ValueObject. We haven't done benchmarking to know that the increased efficiency of field access/memory footprint/etc. is worth the increased complexity. In my experience, stuff like this becomes harder to debug if anything goes wrong. Writing a constructor isn't *that* much boilerplate, and there's also the question of passing derived data to the constructor vs. having the object be responsible for computing it (I think we're doing way too much work outside the object so these constructors end up bloated).

Last @alex and I talked, it seemed that it was a good idea to unify how we treat objects across `isovar`, so this is a step in the right direction - I'm just not sold on this being the approach we want to ultimately unify around :)